library/network/src/lib/network/susefirewall.rb

```diff
@@ -536,21 +536,31 @@ def IsKnownZone(zone)
     # installation)
     #
     # @return [Boolean] whether the selected firewall backend is installed
-    def SuSEFirewallIsInstalled
-      # Always recheck the status in inst-sys, user/solver might have change
-      # the list of packages selected for installation
-      # bnc#892935: in inst_finish, the package is already installed
+    def SuSEFirewallIsSelectedOrInstalled
+      return true if @needed_packages_installed
```

Comment:
Why do we need a cache and why doe it only cache positive results?

Reply:
This method does not cache anything but return true if the package is alrady installed. I suppose it was cached because IsInstalled is called multiple times, so once installed does not make sense to task PackageSystem each time.

In case of the selection, the user can change the package selection so not make sense.